content/40.units_and_quantities.md

```diff
@@ -1,269 +1,642 @@
+is not a base unit, and `Msun / Mpc**3` is a combination of atomic units, which are not in CGS, and
+one of them even has an SI prefix. The dimensions of `g` are `mass` and the CGS conversion factor is
+1. The dimensions of `erg` are `mass * length**2 / time**2` and the CGS conversion factor is 1. The
+dimensions of `Msun / Mpc**3` are `mass / length**3` and the CGS conversion factor is about $6.8
+\times 10^{-41}$.
+
+We use the `UnitRegistry` class to define all valid atomic units. All unit registries contain a unit
+symbol lookup table (in a Python `OrderedDict`) containing for each unit in a tuple the conversion factor to SI units, the dimensionality, the additive offset for the unit, a string representing how the unit should be displayed in plots and other media, and a boolean dictating whether the unit is SI-prefixable. 
+Here is what it would look like with the above units:
 
 ```python
->>> from yt.utilities.physical_constants import
-        boltzmann_constant
->>> print(dd[`temperature']*boltzmann_constant)
-[ 1.28e-12 1.29e-12 1.29e-12 ... 
-  1.63e-12 1.59e-12 1.40e-12] erg
+{ 
+    "g":    (1.0e-3, dimensions.mass, 0.0, r"\rm{g}", True),
+    "erg":  (1.0e-7, dimensions.energy, 0.0, r"\rm{erg}", True),
+    "Msun": (mass_sun_kg, dimensions.mass, 0.0, r"\rm{M}_\odot", False),
+    "pc":   (m_per_pc, dimensions.length, 0.0, r"\rm{pc}", True), 
+    "degC": (1.0, dimensions.temperature, -273.15, r"^\circ\rm{C}", True),
+}
 ```
 
-If a user needs the field in a different unit system, they can quickly convert using `convert_to_units` or `in_units`.
+Note that we only define **atomic** units here. There should be no operations or SI prefixes in the
+registry symbol strings. When we parse non-atomic units like `Msun/Mpc**3`, we use the registry to
+look up the symbols. The unit system in `yt` knows how to handle units like `Mpc` by looking up unit
+symbols with and without prefixes and modify the conversion factor appropriately. Note also that in most cases the "additive offset" will be 0--it applies mainly to temperature scales like Fahrenheit and Celsius where the zero-point of the scale are not equivalent between the two scales, and are also not equivalent to "zero thermal energy", for which the relevant temperature units are Rankine and Kelvin, respectively.
+
+We construct a `Unit` object by providing a string containing atomic unit symbols, combined with operations in Python syntax, and the registry those atomic unit symbols are defined in. We use
+`sympy`'s string parsing features to create the unit expression from the user-provided string. Each `Unit`
+object also has its own unit `sympy` `Expression` object, a dimensionality `Expression` object, a `UnitRegistry` instance, a scalar conversion factor to SI units, additive offset value, and display representation. Among other methods, `Unit` defines the `mul`, `div`, `pow`, and `eq` operations with
```

Comment:
are `Expr` and `Expression` the same class ?

Reply:
Changed it to `Expr`